lib/os/cbprintf_complete.c

```diff
@@ -1815,7 +1815,7 @@ int z_cbvprintf_impl(cbprintf_cb out, void *ctx, const char *fp,
 
 			OUTS(cp, bpe);
 		} else {
-			if (conv->altform_0c | conv->altform_0) {
+			if (conv->altform_0c || conv->altform_0) {
```

Comment:
Are we 100% sure this was a typo and not testing for a non-zero result from bitwise OR?  Deserves an explicit callout in the commit, or to be a commit on its own.  This is a bugfix not a coding style update.

Reply:
yes, both of these values are 1-bit boolean bitfields, and so should be treated as boolean instead of binary.
I agree that a note in the commit would be good to clarify this.

Reply:
If this is really a bugfix, I think moving this change into a extra commit with its own PR would be more appropriate. Than it is not "hidden" in a coding guidelines adaption PR. What do you think?

Reply:
Could go either way - it's not fixing a execution bug as the bit-wise and boolean or operators end up producing the same effect when operating on two boolean values. I can also easily imagine someone writing the bitwise version because it was 'faster'. Whether you consider it a source-level bug seems like a matter of opinion when using the fine C language.

Reply:
That is true. In this case I will move that to an extra commit in this PR to have the explicit callout as andy said. I will also mention it in the PR description.

Reply:
@andyross @keith-packard I moved the change to an own commit and updated the PR description